internal/command/meta_backend.go

```diff
@@ -2533,6 +2586,56 @@ func (m *Meta) GetStateStoreProviderFactory(config *configs.StateStore, locks *d
 	return factory, diags
 }
 
+func (m *Meta) StateStoreProviderFactoryFromConfigState(cfgState *workdir.StateStoreConfigState, locks *depsfile.Locks) (providers.Factory, tfdiags.Diagnostics) {
```

Comment:
nit: Could we make the names of this method and the related method above  -`GetStateStoreProviderFactory`- more similar?

I don't have strong opinions about what the names but given the methods are somewhat related I think matched naming would be nice. Some options:
* `GetStateStoreProviderFactory` & `GetStateStoreProviderFactoryFromConfigState`
* `StateStoreProviderFactory` & `StateStoreProviderFactoryFromConfigState`
* `StateStoreProviderFactoryFromConfig` & `StateStoreProviderFactoryFromConfigState`
* An option I haven't thought of that's better than the above ✨